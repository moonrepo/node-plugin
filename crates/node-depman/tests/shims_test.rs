use proto_pdk_test_utils::{create_plugin, generate_global_shims_test};
use starbase_sandbox::{assert_snapshot, create_empty_sandbox};

#[cfg(not(windows))]
mod npm {
    use super::*;

    generate_global_shims_test!("npm-test", ["node-gyp"]);
}

#[cfg(not(windows))]
mod pnpm {
    use super::*;

    generate_global_shims_test!("pnpm-test", ["pnpx"]);
}

#[cfg(not(windows))]
mod yarn {
    use super::*;

    generate_global_shims_test!("yarn-test", ["yarnpkg"]);
}
