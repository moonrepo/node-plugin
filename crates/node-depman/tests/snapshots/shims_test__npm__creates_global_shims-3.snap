---
source: crates/node-depman/tests/shims_test.rs
expression: "std::fs::read_to_string(sandbox.path().join(\".proto/shims\").join(if cfg!(windows)\n                    {\n                    format!(\"{}.cmd\", \"node-gyp\")\n                } else { \"node-gyp\".to_string() })).unwrap()"
---
#!/usr/bin/env bash
set -e
[ -n "$PROTO_DEBUG" ] && set -x


exec proto run npm-test --alt "node-gyp" --  "$@" 


